### ISSUE TITLE
Mixed Content errors over AJAX.

### ISSUE THREAD

**Issue by agpar** [DDMAL/misirlou#41]
I have celery up and running now, and tried importing a manifest on the live server. We get a number of Mixed Content errors regarding making requests to http endpoints from our https site. We even get one for polling our own server using http!

So, this raises a number of pretty thorny issues. Maybe it is best to only retrieve manifests through solr, as our project's basic function will be dependent on everyone else keeping their certs up to date otherwise.

Note, it looks like we can import a manifest hosted at a non-secure endpoint (since the doc is only retrieved by Celery), but we run into  issues displaying it (since we're making an ajax request to a non-secure endpoint). 

Interested in your thoughts about how to handle this!

**Comment by ahankinson** [comment]
All of the requests to our own server should be over https.

Caching external manifests sounds like a good thing to do; we can periodically poll and checksum to see if they've changed and update them.

**Comment by agpar** [comment]
I think @wabain will have an easy time fixing this. It should only be a few changes to requests on the frontend.

**Comment by agpar** [comment]
a538d0d852b86860d80f62f7bd68e76b4e031484 fixes the problem of django returning `http` urls when using `reverse()`. 767b3cf80b717158e80411424e37625930d56163 makes it possible to retrieve indexed manifests from our own server. We should be in a better position to tackle these problems now.

**Comment by agpar** [comment]
I added a `/images/` endpoint on dev to proxy insecure images (27d9dda4b8a38afe626146ed74b67cd4014010a7). We can get the manifests from our own server, then proxy through the images if their certs are out of date or non-existent. `/images/?img=[URL]` will just get you the image at the URL. I threw this together rather quickly, with the thought that mixed content blocks on images could be an issue.

This is based on a code snippet from DIAMM, only theirs used pk's from their DB to reference images, rather than raw urls. I think what I've implemented has potential to be abused (although, what good would it do other people to secure images through our site?). 

We could do some sort of session system, where a second `pk` key needs to be passed in. This key could be used to populate a cache with a set of valid image urls for the manifest being referenced. Then we make sure we only serve images contained in manifests we serve. Not sure if this is necessary.
